src/components/editor/Editor.tsx

```diff
@@ -106,7 +100,16 @@ class Editor extends React.PureComponent<Partial<IEditorProps>,
   }
 
   isNewAnnotation() {
-    return this.props.editor.annotationId !== null;
+    return this.props.annotationId !== null;
+  }
+
+  onDrag = (delta: IVec2) => {
```

Comment:
Couldn't this be handled inside `DraggableWidget` ?

Reply:
Wouldn't it be enough to:
- keep `locationX` and `locationY` as `DraggableWidget` state (btw. I think this could be squashed to `location: { x, y }`, as any change of either x or y must result in a rerender - and we won't have to pass two fields through some of the components who don't care about them)
- move the `deriveStateFromProps` logic to the constructor (so it's only set when `DraggableWidget` is created - from the initial `x` and `y` coords passed from `Editor` or any other future component)
- (optional, I guess?) remove `componentDidUpdate()` from `DraggableWidget` (unless recreating `dragTracker` is necessary, but I highly doubt it?)

Reply:
Widget location is kept in the global `state`. If it changes, it's passed to Editor. How can we keep Editor from overwriting (by passing on the location) the location DraggableWidget tries to maintain?

Reply:
as discussed on Slack - if it's the first render of `DraggableWidget` - only `constructor` (and later `componentDidUpdate`, but constructor will be fine I think) will be invoked.
If it's a further update - `componentDidUpdate` will be invoked.

However for this to work, I'd propose to move the `visible` flag handling from `Widget` to `Editor` (a simple `if (!visible) return null` would be sufficient at the beginning of render fn) - so that `DraggableWidget` is actually recreated each time visible is changed from false to true.

On second thought, actually dismounting the whole `Editor` component and mounting it later on according to `visible` status would be even better, but then we'd have to introduce some kind of `WidgetManager`, which would handle stuff like this (i.e. conditional rendering of `Editor`, `Viewer`, `Menu` etc, instead of them being always render in `App`)

Reply:
Moved to roadmap (`miscallaneous`)